### PR TITLE
Fix/parallel space calc

### PR DESCRIPTION
**Pull request by ianopolous**
Parallelise space recalculation

### CODE REVIEW

**Comment by ianopolous** [comment]
Can confirm this reduces the restart time on the demo server to 58s.

**Comment by ianopolous** [comment]
Ok this is good to go now.

File: src/peergos/server/SpaceCheckingKeyFilter.java [hunk 1]
```diff
@@ -113,14 +114,30 @@ public SpaceCheckingKeyFilter(CoreNode core,
     private void loadAllOwners() {
         try {
             List<String> usernames = core.getUsernames("").get();
+            if (usernames.size() == 0)
+                return;
+            int threads = Math.min(usernames.size(), 1000);
```
Comment (cboddy) [inline src/peergos/server/SpaceCheckingKeyFilter.java]: Why 1000 threads?

File: src/peergos/shared/util/Exceptions.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package peergos.shared.util;
+
+import java.util.concurrent.*;
+
+public class Exceptions {
```
Comment (cboddy) [inline src/peergos/shared/util/Exceptions.java]: test?

Comment (ianopolous) [reply src/peergos/shared/util/Exceptions.java]: yes I have

**Pull request merged**